### ISSUE TITLE
Site editor: the header toolbar should be a ARIA toolbar

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#41869]
### Description

In the _block_ editor, the 'Document tools' toolbar in the header is implemented as a ARIA toolbar.

<img width="1792" alt="block editor toolbar" src="https://user-images.githubusercontent.com/1682452/175010714-0c80a2cf-0d4c-422d-9f3f-21e8d0da63eb.png">

- Semantically, the attributes `role="toolbar" aria-orientation="horizontal" aria-label="Document tools"` provide the necessary information and labelling so that assistive technology users are aware it's a toolbar with a specific keyboard interaction model.
- Navigation with arrow/Home/End keys helps keyboard users by reducing the amount ot Tab stops in the interface. The toolbar itself has only one focusable element at a time. Keyboard users can move to other elements in the toolbar by using the arrow/Home/End keys.

Overall, it's a very good implementation of [the Toolbar ARIA pattern](https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/).

In the _site_ editor, the header toolbar looks the same but it doesn't provide any of the above functionalities:

<img width="1792" alt="site editor toolbar" src="https://user-images.githubusercontent.com/1682452/175012848-b4cefe3e-a42c-4868-89f6-80d24240760c.png">

- No ARIA toolbar semantics.
- No arrow/Home/End keys navigation.

Visually the two toolbar are very similar and users would expect they work in the same way. However, the semantics and interaction are inconsistent.

Thee two toolbars should provide users with the same functionality, interaction model, and accessibility level.

### Step-by-step reproduction instructions

- Go to the block editor.
- Use the Tab key to navigate the interface.
- Observe the header toolbar has only one tab stop.
- Once the header toolbar has focus, observe the arrow keys and the Home/End keys allow to navigate within the toolbar.
- Go to the site editor.
- Use the Tab key to navigate the interface.
- Observe that each control within the header toolbar is a tab stop.
- Observe that arrows/Home/End keys navigation within the toolbar doesn't work.

### Environment info

Wordpress trunk 6.1
Gutenberg plugin Version 13.5.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
